exporter/metric/constants.go

```diff
@@ -23,45 +23,75 @@ package metric
+	ServiceKeyNamespace = "service.namespace"
+	// Deprecated: use semconv.ServiceInstanceIDKey instead
 	ServiceKeyInstanceID = "service.instance.id"
-	ServiceKeyName       = "service.name"
+	// Deprecated: use semconv.ServiceNameKey instead
+	ServiceKeyName = "service.name"
 
+	// Deprecated: HostType is not needed.
 	HostType = "host"
 	// A uniquely identifying name for the host.
+	// Deprecated: use semconv.HostNameKey instead
 	HostKeyName = "host.name"
 	// A hostname as returned by the 'hostname' command on host machine.
+	// Deprecated: HostKeyHostName is not needed.
 	HostKeyHostName = "host.hostname"
-	HostKeyID       = "host.id"
-	HostKeyType     = "host.type"
+	// Deprecated: use semconv.HostIDKey instead
+	HostKeyID = "host.id"
+	// Deprecated: use semconv.HostTypeKey instead
+	HostKeyType = "host.type"
 
 	// A uniquely identifying name for the Container.
-	ContainerKeyName      = "container.name"
+	// Deprecated: use semconv.ContainerNameKey instead
+	ContainerKeyName = "container.name"
+	// Deprecated: use semconv.ContainerImageNameKey instead
 	ContainerKeyImageName = "container.image.name"
-	ContainerKeyImageTag  = "container.image.tag"
+	// Deprecated: use semconv.ContainerImageTagKey instead
+	ContainerKeyImageTag = "container.image.tag"
 
 	// Cloud Providers
-	CloudProviderAWS   = "aws"
-	CloudProviderGCP   = "gcp"
+	// Deprecated: use semconv.CloudProviderAWS instead
+	CloudProviderAWS = "aws"
+	// Deprecated: use semconv.CloudProviderGCP instead
+	CloudProviderGCP = "gcp"
+	// Deprecated: use semconv.CloudProviderAzure instead
 	CloudProviderAZURE = "azure"
 
-	K8S                  = "k8s"
-	K8SKeyClusterName    = "k8s.cluster.name"
-	K8SKeyNamespaceName  = "k8s.namespace.name"
-	K8SKeyPodName        = "k8s.pod.name"
+	// Deprecated: Use "k8s" instead. This should not be needed
+	K8S = "k8s"
+	// Deprecated: use semconv.K8SClusterNameKey instead
+	K8SKeyClusterName = "k8s.cluster.name"
+	// Deprecated: use semconv.K8SNamespaceNameKey instead
+	K8SKeyNamespaceName = "k8s.namespace.name"
+	// Deprecated: use semconv.K8SPodNameKey instead
+	K8SKeyPodName = "k8s.pod.name"
+	// Deprecated: use semconv.K8SDeploymentNameKey instead
 	K8SKeyDeploymentName = "k8s.deployment.name"
 
 	// Monitored Resources types
-	K8SContainer   = "k8s_container"
-	K8SNode        = "k8s_node"
-	K8SPod         = "k8s_pod"
-	K8SCluster     = "k8s_cluster"
-	GCEInstance    = "gce_instance"
+	// Deprecated: Use "k8s_container" instead
+	K8SContainer = "k8s_container"
+	// Deprecated: Use "k8s_node" instead
+	K8SNode = "k8s_node"
+	// Deprecated: Use "k8s_pod" instead
+	K8SPod = "k8s_pod"
+	// Deprecated: Use "k8s_cluster" instead
+	K8SCluster = "k8s_cluster"
+	// Deprecated: Use "gce_instance" instead
+	GCEInstance = "gce_instance"
+	// Deprecated: Use "aws_ec2_instance" instead
 	AWSEC2Instance = "aws_ec2_instance"
-	GenericTask    = "generic_task"
+	// Deprecated: Use "generic_task" instead
+	GenericTask = "generic_task"
```

Comment:
why are we deprecating these constants to tell users to use the strings instead? Can't we keep providing these constants?

Reply:
There really isn't a reason (I can think of) why someone would need to use these constants with the exporter, which is why I'd prefer not to keep them.  If someone does need that string for some reason, they can just define the constant themselves.

If we ever support custom monitored resource mappings (which might be a useful feature), we would re-introduce similar constants at that point.